lib/SILOptimizer/Mandatory/RawSILInstLowering.cpp

```diff
@@ -286,17 +204,23 @@ lowerAssignOrInitInstruction(SILBuilderWithScope &b,
       CanSILFunctionType fTy = initFn->getType().castTo<SILFunctionType>();
       SILFunctionConventions convention(fTy, inst->getModule());
 
-      auto selfValue = inst->getSelf();
-      auto isRefSelf = selfValue->getType().getASTType()->mayHaveSuperclass();
-
-      SILValue selfRef;
-      if (isRefSelf) {
-        selfRef = b.emitBeginBorrowOperation(loc, selfValue);
-      } else {
-        selfRef = b.createBeginAccess(loc, selfValue, SILAccessKind::Modify,
-                                      SILAccessEnforcement::Dynamic,
-                                      /*noNestedConflict=*/false,
-                                      /*fromBuiltin=*/false);
+      auto inLocalContext =
+          inst->getProperty()->getDeclContext()->isLocalContext();
+      auto selfOrLocalValue = inst->getSelfOrLocalOperand();
+      bool isRefSelf =
+          selfOrLocalValue->getType().getASTType()->mayHaveSuperclass();
+
+      SILValue selfRef = nullptr;
+      if (!inLocalContext) {
+        if (isRefSelf) {
```

Comment:
I think this should be done right before https://github.com/swiftlang/swift/pull/83886/files#diff-8a20b78e30c1606aca553dafc619738e1dd72b952ed5ba1c07e82b8fb2c6abe0R251 and `emitFieldRerference` should make `SILValue selfRef` as an argument.